src/test/java/org/manifold/compiler/front/TestExpressionGraph.java

```diff
@@ -36,4 +39,61 @@ public void testAddEdge_targetNotInGraph_exceptionThrown() {
     expressionGraph.addVertex(from);
     expressionGraph.addEdge(edgeOfTomCruise);
   }
+
+
+  @Test
+  public void testAddSubExpressionGraph_correctGraphAdded() throws Exception {
+    // init main graph
+    ExpressionGraph mainGraph = new ExpressionGraph();
+
+    // input/output
+    VariableIdentifier mainInput = new VariableIdentifier(ImmutableList.of("mainInput"));
+    VariableIdentifier mainOutput = new VariableIdentifier(ImmutableList.of("mainOutput"));
+    mainGraph.addVertex(mainInput);
+    mainGraph.addVertex(mainOutput);
+
+    ExpressionVertex mainInputVertex = mainGraph.getVariableVertex(mainInput);
+    ExpressionVertex mainOutputVertex = mainGraph.getVariableVertex(mainOutput);
+
+    // function call vertex
+    /* this doesn't currently work due to a potential bug in FunctionTypeValueVertex where source is set instead of
+       target, but the test doesn't really need it (since in actual use we'd remove this vertex + its edges before
+       replacing it with the subgraph
+    ExpressionEdge inputEdge = new ExpressionEdge(mainInputVertex, null);
+    ExpressionEdge outputEdge = new ExpressionEdge(null, mainOutputVertex);
+    ExpressionVertex function = new FunctionTypeValueVertex(mainGraph, inputEdge, outputEdge);
```

Comment:
This isn't actually the function call; this is only the vertex that tells you what type your function has.